### PR TITLE
ConstantTVDomain typing issue fix

### PR DESCRIPTION
**Pull request by hwpang**
This PR aims to resolve the typing issue of ConstantTVDomain mentioned in issue #84 .

It includes
- change multiple keyword variables to position variables in getkfkrevs to allow dispatch
- add a method of getkfkrevs to deal with the type change related to autodiff p

### CODE REVIEW

File: src/PhaseState.jl [hunk 1]
```diff
@@ -190,4 +190,26 @@ export getkfkrev
     end
     return kfs,krev
 end
+@inline function getkfkrevs(phase::U,T::W1,P::W2,C::W3,N::W4,ns::Q1,Gs::Array{Q2,1},diffs::Q3,V::W5;kfs::W6=nothing) where {U<:AbstractPhase,W6,W5<:Real,W1<:Real,W2<:Real,W3<:Real,W4<:Real, Q1<:AbstractArray,Q2<:ForwardDiff.Dual,Q3<:AbstractArray} #autodiff p
```
Comment (mjohnson541) [inline src/PhaseState.jl:193]: Is there a better way to type these so they work properly with reverse mode differentiation (that might send something through that isn't an Array of ForwardDiff.Dual) as well as ForwardDiff?

Comment (hwpang) [reply src/PhaseState.jl]: Thanks for the comment! I will dispatch getkfkrevs for both ReverseDiff and Tracker TrackedArray.

**Comment by hwpang** [comment]
> This looks pretty good, just one question.

Thanks for the comment & review! I dispatch the getkfkrevs into three methods, i.e. one for ForwardDiff autodiff p, another for ReverseDiff autodiff p, and the other for autodiff y + non-autodiff. I tested by simulating ConstantTVDomain and using getadjointsensitivities, and it agrees with forward sensitivities.

**Comment by mjohnson541** [comment]
Can we add a test for adjoint sensitivities for ConstantTVDomain?

**Comment by mjohnson541** [comment]
Decided offline to wait on the test until we have analytic parameter jacobians.

**Pull request merged**